database/sqlite/sqlite_health.c

```diff
@@ -358,24 +359,51 @@ void sql_health_alarm_log_save(RRDHOST *host, ALARM_ENTRY *ae)
 }
 
 /* Health related SQL queries
-   Cleans up the health_log table.
+   Get a count of rows from health log table
 */
-#define SQL_CLEANUP_HEALTH_LOG(guid,guid2,limit) "DELETE from health_log_%s where unique_id in (SELECT unique_id from health_log_%s order by unique_id asc LIMIT %lu);", guid, guid2, limit
-void sql_health_alarm_log_cleanup(RRDHOST *host) {
+#define SQL_COUNT_HEALTH_LOG(guid) "SELECT count(1) FROM health_log_%s;", guid
+void sql_health_alarm_log_count(RRDHOST *host) {
     sqlite3_stmt *res = NULL;
-    static size_t rotate_every = 0;
     int rc;
     char command[MAX_HEALTH_SQL_SIZE + 1];
 
-    if(unlikely(rotate_every == 0)) {
-        rotate_every = (size_t)config_get_number(CONFIG_SECTION_HEALTH, "rotate log every lines", 2000);
-        if(rotate_every < 100) rotate_every = 100;
+    if (unlikely(!db_meta)) {
+        if (default_rrd_memory_mode == RRD_MEMORY_MODE_DBENGINE)
+            error_report("Database has not been initialized");
+        return;
     }
 
-    if(likely(host->health.health_log_entries_written < rotate_every)) {
+    char uuid_str[GUID_LEN + 1];
+    uuid_unparse_lower_fix(&host->host_uuid, uuid_str);
+
+    snprintfz(command, MAX_HEALTH_SQL_SIZE, SQL_COUNT_HEALTH_LOG(uuid_str));
+
+    rc = sqlite3_prepare_v2(db_meta, command, -1, &res, 0);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to prepare statement to count health log entries from db");
         return;
     }
 
+    rc = sqlite3_step_monitored(res);
+    if (likely(rc == SQLITE_ROW))
+        host->health.health_log_entries_written = (size_t) sqlite3_column_int64(res, 0);
+
+    rc = sqlite3_finalize(res);
+    if (unlikely(rc != SQLITE_OK))
+        error_report("Failed to finalize the prepared statement to count health log entries from db");
+
+    info("HEALTH [%s]: Table health_log_%s, contains %lu entries.", rrdhost_hostname(host), uuid_str, (unsigned long int) host->health.health_log_entries_written);
+}
+
+/* Health related SQL queries
+   Cleans up the health_log table on a non-claimed host
+*/
+#define SQL_CLEANUP_HEALTH_LOG_NOT_CLAIMED(guid,guid2,limit) "DELETE from health_log_%s where unique_id in (SELECT unique_id from health_log_%s order by unique_id asc LIMIT %lu);", guid, guid2, limit
```

Comment:
Well, we are removing data from table here, so my suggestion is to simplify engine life removing `order by unique_id`.

Reply:
I will update the comment here, because it loos like I missed something before our meeting.

I was reviewing line 455 where you again order indexes that will be excluded. I arrived in the conclusion you are doing this o speed up search on index, if this is true, why are we ordering indexes in ASC order when we are in normal situation searching the most recent? :thinking: 

If the reason you are ordering is to simplify the search on index, we need to reverse our index order.

Reply:
Thanks Thiago. What this tries to do is to delete older entries from the table, leaving in there the latest (by default) 2000 entries. This statement has worked for a while too. The change here is that we will continue to do so when the agent is un-claimed, and follow a different strategy to cleanup the table when it's claimed. We can discuss if you want if I didn't understand the question.

Reply:
It called my attention the fact that we order to exclude. The unique possible logic to do this that came in my mind is to speed up the exclusion by database engine.

Now the issue, if we order in `asc` instead `desc`, this means that our indexes were ordered in a way that we are not given priority for the latest data, but netdata is a monitoring tool that has a RRD database, so the most recent data should be prioritized.  This was the contradiction that query was showing.

The query is not wrong, this is not the subject of my comments. The possible issue is the optimization of the queries we run.

Reply:
Thanks guys, changed the query.